File: keyphrase-extraction/microsoft-cognitive-services-text-analytics/MicrosoftCognitiveServicesTextAnalyticsKeyphraseExtraction.java
public class Document {
    public String id, language, text;

    public Document(String id, String language, String text){
        this.id = id;
        this.language = language;
        this.text = text;
    }
}

public class Documents {
    public List<Document> documents;

    public Documents() {
        this.documents = new ArrayList<>();
    }
    public void add(String id, String language, String text) {
        this.documents.add (new Document (id, language, text));
    }
}

public class KeyphraseExtraction {
    public String id;
    public List<String> keyPhrases;
}

public class KeyphraseExtractionResponse {
    public List<KeyphraseExtraction> documents;
}

String host = "https://westeurope.api.cognitive.microsoft.com";
String path = "/text/analytics/v2.0/keyPhrases";
String url = host.concat(path);

Documents documents = new Documents();
documents.add(id, language, text);

Gson gson = new Gson();
String body = gson.toJson(documents);

String stringResponse = Unirest.post(url)
    .header("Content-Type", "application/json")
    .header("Ocp-Apim-Subscription-Key", subscriptionKey)
    .body(body)
    .asString()
    .getBody();

KeyphraseExtractionResponse response = gson.fromJson(stringResponse, KeyphraseExtractionResponse.class);

File: keyphrase-extraction/microsoft-cognitive-services-text-analytics/MicrosoftCognitiveServicesTextAnalyticsKeyphraseExtraction.kt
// "text" is a String containg the input document.
// For English, set "language" to "en".

class Document(val id: String, val language: String, var text: String)

data class KeyphraseExtraction(
    val id: String,
    val keyPhrases: List<String>)

data class KeyphraseExtractionResponse(
    val documents: List<KeyphraseExtraction>)

val host = "https://westeurope.api.cognitive.microsoft.com"
val path = "/text/analytics/v2.0/keyPhrases"
val url = host + path

val documents = listOf(Document("1", language, text))

val gson = Gson()
val body = gson.toJson(documents)

val stringResponse = Unirest.post(url)
    .header("Content-Type", "application/json")
    .header("Ocp-Apim-Subscription-Key", subscriptionKey)
    .body(body)
    .asString()
    .body

val response = gson.fromJson(stringResponse, KeyphraseExtractionResponse::class.java)
val keyPhrases = response.documents.single().keyPhrases
